cli/groups_test.go

```diff
@@ -0,0 +1,1005 @@
+	cases := []struct {
+		desc          string
+		args          []string
+		logType       outputLog
+		group         mgsdk.Group
+		sdkErr        errors.SDKError
+		errLogMessage string
+	}{
+		{
+			desc: "create group successfully",
+			args: []string{
+				crtCmd,
+				groupJson,
+				token,
+			},
+			group:   group,
+			logType: entityLog,
+		},
+		{
+			desc: "create group with invalid args",
+			args: []string{
+				crtCmd,
+				groupJson,
+				token,
+				extraArg,
+			},
+			logType: usageLog,
+		},
+		{
+			desc: "create group with invalid json",
+			args: []string{
+				crtCmd,
+				"{\"name\":\"testgroup\", \"metadata\":{\"key1\":\"value1\"}",
+				token,
+			},
+			sdkErr:        errors.NewSDKError(errors.New("unexpected end of JSON input")),
+			errLogMessage: fmt.Sprintf("\nerror: %s\n\n", errors.New("unexpected end of JSON input")),
+			logType:       errLog,
+		},
+		{
+			desc: "create group with invalid token",
+			args: []string{
+				crtCmd,
+				groupJson,
+				invalidToken,
+			},
+			sdkErr:        errors.NewSDKErrorWithStatus(svcerr.ErrAuthorization, http.StatusUnauthorized),
+			errLogMessage: fmt.Sprintf("\nerror: %s\n\n", errors.NewSDKErrorWithStatus(svcerr.ErrAuthorization, http.StatusUnauthorized)),
+			logType:       errLog,
+		},
+		{
+			desc: "create group without domain token",
+			args: []string{
+				crtCmd,
+				groupJson,
+				tokenWithoutDomain,
+			},
+			sdkErr:        errors.NewSDKErrorWithStatus(svcerr.ErrDomainAuthorization, http.StatusForbidden),
+			errLogMessage: fmt.Sprintf("\nerror: %s\n\n", errors.NewSDKErrorWithStatus(svcerr.ErrDomainAuthorization, http.StatusForbidden)),
+			logType:       errLog,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			sdkCall := sdkMock.On("CreateGroup", mock.Anything, tc.args[2]).Return(tc.group, tc.sdkErr)
+			out := executeCommand(t, rootCmd, tc.args...)
+
+			switch tc.logType {
+			case entityLog:
+				err := json.Unmarshal([]byte(out), &gp)
+				assert.Nil(t, err)
+				assert.Equal(t, tc.group, gp, fmt.Sprintf("%s unexpected response: expected: %v, got: %v", tc.desc, tc.group, gp))
+			case usageLog:
+				assert.False(t, strings.Contains(out, rootCmd.Use), fmt.Sprintf("%s invalid usage: %s", tc.desc, out))
+			case errLog:
+				assert.Equal(t, tc.errLogMessage, out, fmt.Sprintf("%s unexpected error response: expected %s got errLogMessage:%s", tc.desc, tc.errLogMessage, out))
+			}
+			sdkCall.Unset()
+		})
+	}
+}
+
+func TestGetGroupsCmd(t *testing.T) {
+	sdkMock := new(sdkmocks.SDK)
+	cli.SetSDK(sdkMock)
+	groupCmd := cli.NewGroupsCmd()
+	rootCmd := setFlags(groupCmd)
+
+	var ch mgsdk.Group
+	var page mgsdk.GroupsPage
+
+	cases := []struct {
+		desc          string
+		args          []string
+		sdkErr        errors.SDKError
+		page          mgsdk.GroupsPage
+		group         mgsdk.Group
+		logType       outputLog
+		errLogMessage string
+	}{
+		{
+			desc: "get all groups successfully",
+			args: []string{
+				getCmd,
```

Comment:
There is no need to pass a command always as a first arg (since it's the same for all the cases); instead, use `execCmd ` like this:
```go
out := execCmd(t, rootCmd, append([]string{createCmd}, tc.args...)...)
```